test/js/node/test/parallel/test-path-resolve.js

```diff
@@ -0,0 +1,97 @@
+     [['d:/ignore', 'd:some/dir//'], 'd:\\ignore\\some\\dir'],
+     [['.'], process.cwd()],
+     [['//server/share', '..', 'relative\\'], '\\\\server\\share\\relative'],
+     [['c:/', '//'], 'c:\\'],
+     [['c:/', '//dir'], 'c:\\dir'],
+     [['c:/', '//server/share'], '\\\\server\\share\\'],
+     [['c:/', '//server//share'], '\\\\server\\share\\'],
+     [['c:/', '///some//dir'], 'c:\\some\\dir'],
+     [['C:\\foo\\tmp.3\\', '..\\tmp.3\\cycles\\root.js'],
+      'C:\\foo\\tmp.3\\cycles\\root.js'],
+      // IMPORTANT NOTE: 
+      // - PR originally landed in #54224 and #55623 to fix issue #54025
+      // - It caused a regression (issue #56002) and was reverted in #56088
+      // - This behavior did _not_ land in even-numbered versions
+      // If node decides to adopt this, we need to revisit these tests
+      // 
+      //  [['\\\\.\\PHYSICALDRIVE0'], '\\\\.\\PHYSICALDRIVE0'],
+      //  [['\\\\?\\PHYSICALDRIVE0'], '\\\\?\\PHYSICALDRIVE0'],
+     [['\\\\.\\PHYSICALDRIVE0'], '\\\\.\\PHYSICALDRIVE0\\'],
+     [['\\\\?\\PHYSICALDRIVE0'], '\\\\?\\PHYSICALDRIVE0\\'],
+    ],
+  ],
+  [ path.posix.resolve,
+    // Arguments                    result
+    [[['/var/lib', '../', 'file/'], '/var/file'],
+     [['/var/lib', '/../', 'file/'], '/file'],
+     [['a/b/c/', '../../..'], posixyCwd],
+     [['.'], posixyCwd],
+     [['/some/dir', '.', '/absolute/'], '/absolute'],
+     [['/foo/tmp.3/', '../tmp.3/cycles/root.js'], '/foo/tmp.3/cycles/root.js'],
+    ],
+  ],
+];
+resolveTests.forEach(([resolve, tests]) => {
+  tests.forEach(([test, expected]) => {
+    const actual = resolve.apply(null, test);
+    let actualAlt;
+    const os = resolve === path.win32.resolve ? 'win32' : 'posix';
+    if (resolve === path.win32.resolve && !common.isWindows)
+      actualAlt = actual.replace(backslashRE, '/');
+    else if (resolve !== path.win32.resolve && common.isWindows)
+      actualAlt = actual.replace(slashRE, '\\');
+
+    const message =
+      `path.${os}.resolve(${test.map(JSON.stringify).join(',')})\n  expect=${
+        JSON.stringify(expected)}\n  actual=${JSON.stringify(actual)}`;
+    if (actual !== expected && actualAlt !== expected)
+      failures.push(message);
+  });
+});
+assert.strictEqual(failures.length, 0, failures.join('\n'));
+
+if (common.isWindows) {
+  // Test resolving the current Windows drive letter from a spawned process.
+  // See https://github.com/nodejs/node/issues/7215
+  const currentDriveLetter = path.parse(process.cwd()).root.substring(0, 2);
+  const resolveFixture = fixtures.path('path-resolve.js');
+  const spawnResult = child.spawnSync(
+    process.argv[0], [resolveFixture, currentDriveLetter]);
+  const resolvedPath = spawnResult.stdout.toString().trim();
+  assert.strictEqual(resolvedPath.toLowerCase(), process.cwd().toLowerCase());
+}
+
+// TODO
+// if (!common.isWindows) {
+//   // Test handling relative paths to be safe when process.cwd() fails.
+//   process.cwd = () => '';
```

Comment:
i think it is ok to not support monkey-patching the process.cwd function for `node:path` resolution

Reply:
After trying various methods to remove `cwd`, I've commented it out and added an explainer.